Add Actions for VLC, iTunes, Pixelmator, Sublime Text.  Add Volume control to Spotify

ejwill opened the pull request:
Add play/pause, previous/next track, volume increase/decrease to VLC and iTunes.  Created default.json for both VLC and iTunes.  See Pixelmator and Sublime file names for what actions are covered.  Added volume increase/decrease control to Spotify and updated Spotify default.json.  I uploaded Pixelmator and Sublime as script files, all others are text files.

ejwill committed:
brightness and space changing

Added Apple Scripts for increasing/decreasing the brightness and moving
from one space to the next left or right

ejwill committed:
Revert "brightness and space changing"

This reverts commit c26edc1d25a7c4026ba9f5dce1166099263b8841.

ejwill committed:
Move Space and Brightness

created applescripts to move space left or right and increase/decrease
brightness

ejwill committed:
update move spaces file name

ejwill committed:
Add Actions for Moving Spaces and Changing Brightness for System (#11)

Added Apple Scripts for increasing/decreasing the brightness and moving
from one space to the next left or right

ejwill committed:
Add VLC and iTunes, Volume for Spotify

Add play/pause, previous/next track, volume increase/decrease to VLC
and iTunes.  Created default.json for both VLC and iTunes.  Added
volume increase/decrease control to Spotify.  Updated Spotify
default.json

ejwill committed:
Merge remote-tracking branch 'getsenic/master'

ejwill committed:
bunch of Pixelmater actions

many different actions; see file names for what they do.  Files are
applescript script files can upload text files instead if I have to

ejwill committed:
sublime text actions

added numerous sublime text actions.  files saved as script files.

larsblumberg commented:
Thanks a lot for your Pull Request! Can you please submit all script files with `.scpt` extension? Some of your files are saved as compiled AppleScripts – can you please save them in plain text? You can open the files with a text editor and paste the raw AppleScript code if needed.

ejwill committed:
Merge remote-tracking branch 'getsenic/master'

ejwill committed:
updated applescript text files to script files

doing what Lars asked in pull request

ejwill commented:
@larsblumberg updated the file types

ejwill committed:
Scrolling for Chrome, Additional Actions for Firefox, Added Safari

Chrome, Firefox, and Safari now have all the same actions.  Scrolling
may still need work; depends how people like it.

ejwill commented:
added actions for Chrome, Firefox, and Safari.  They all have the same actions now.  I did get scrolling to work but it might need tweaks

larsblumberg commented on .gitignore:
```diff
@@ -0,0 +1,6 @@
+
+Application Actions/System/1.0/Decrease Brightness.applescript
+
+Application Actions/System/1.0/Decrease Brightness.applescript
```
Can you please remove this `.gitignore` file?

larsblumberg commented:
Dear @ejwill, the `.scpt` files are stored in binary format – Can you please save them in plain text format? Otherwise we cannot read the AppleScript source code to review it. Thank you!

ejwill committed:
remove gitignore

ejwill committed:
Merge remote-tracking branch 'getsenic/master'

# Conflicts:
#	Application Actions/Google Chrome/1.0/default.json

ejwill commented:
@larsblumberg so how do you want me to save the files then as .applescript since the files have to be in .scpt to work with Numio right?  Do you want me to upload them as .applescript then you change them to .scpt or me to save them as .applescript?

larsblumberg commented:
Right now the Nuimo for OS X App only looks for `.scpt` files. I suggest you to save your `.scpt` files with a text editor to save them in plain text. AppleScript Editor seems to save these files in compiled format. For security reasons, we don't accept compiled files to be uploaded.

ejwill committed:
Delete Decrease Volume script that was for VLC

ejwill committed:
Actions for Firefox

ejwill committed:
Scroll for Chrome

ejwill committed:
Actions for iTunes

ejwill committed:
actions for Pixelmatr

ejwill committed:
Actions for Safari

ejwill committed:
Volume actions for Spotify

ejwill committed:
Actions for Sublime Text

ejwill committed:
Actions for VLC

ejwill commented:
@larsblumberg updated all of the script files to be human readable.  Sorry for all the commits.

ejwill committed:
fixing default.json for Chrome

ejwill committed:
Re adding System volume control

somehow system volume control got removed.  Re-adding files from Nuimo
directory.

hansmbakker commented:
Great work!

One comment: I see in https://github.com/getsenic/nuimo-app-osx-actions/pull/12/files#diff-d34500519472862474d2d399be9afa01R7 that you removed the go back/forward actions from chrome's defaults, but left two pairs of "select previous/next tab".

I'd suggest that Senic decides which defaults there are, and to have them consistent with similar other apps, and I'd suggest not to have duplicated assignments by default removing other (more often used) from the defaults.

larsblumberg commented on Application Actions/Spotify/1.0/Increase Volume.scpt:
```diff
@@ -0,0 +1,8 @@
+tell application "Spotify"
+	if (sound volume < 90) then
+		set the sound volume to (sound volume + 2)
+	else if (sound volume > 90) and (sound volume <= 100) then
```
It should cover the case `sound volume >= 90` here as right now sound volume = 90 isn't covered.

ejwill committed:
cover sound volume 90 for spotify

ejwill committed:
cover sound volume 10 Spotify decrease volume

ejwill commented:
@larsblumberg made an update to Spotify increase and decrease volume to handle sound volume at 90 and 10 respectively.  Also, is there any way to show an icon on the device when performing and actions such as increase/decrease volume

ejwill commented:
@wind-rider I thought next/previous tab was better suited as a default than forward/back.  Could put the go forward/backward on the fly commands.

larsblumberg commented on Application Actions/Firefox/1.0/Scroll Down.scpt:
```diff
@@ -0,0 +1,5 @@
+tell application "System Events" to tell process "Firefox"
```
There's no need to include `to tell process "Firefox"` since key codes can only be sent to the active app.
You can remove `to tell process XYZ` from all other scripts as well.

larsblumberg commented on Application Actions/Firefox/1.0/default.json:
```diff
@@ -1,11 +1,11 @@
+  "RotateLeft"	:	"Scroll Down",
+  "RotateRight"	:	"Scroll Up",
+  "FlyLeft"		:	"Select Previous Tab",
+  "FlyRight"	:	"Select Next Tab",
```
Can you please preserve the formatting? The actions shall be left-aligned. Please don't use tabs but spaces.

larsblumberg commented on Application Actions/Sublime Text/1.0/Redo.scpt:
```diff
@@ -0,0 +1,3 @@
+tell application "System Events" to tell process "Sublime Text"
+	key code 16 using {command down}
```
key code 16 doesn't work with all keyboard layouts. It works with US/English layout but not on German layouts. I guess we shouldn't include these actions then – unless we find a way to click the Undo/Redo menu item from the "Edit" menu – that should be keyboard independent.

larsblumberg commented on Application Actions/Sublime Text/1.0/Toggle Block Comment.scpt:
```diff
@@ -0,0 +1,3 @@
+tell application "System Events" to tell process "Sublime Text"
+	key code 44 using {command down, option down}
```
This key code is also keyboard layout dependent – Any chance we can trigger a menu item click instead?

larsblumberg commented on Application Actions/Sublime Text/1.0/Indent.scpt:
```diff
@@ -0,0 +1,3 @@
+tell application "System Events" to tell process "Sublime Text"
+	key code 30 using command down
```
This key code is also keyboard layout dependent – Any chance we can trigger a menu item click instead?

larsblumberg commented on Application Actions/Sublime Text/1.0/Left Through Tabs.scpt:
```diff
@@ -0,0 +1,3 @@
+tell application "System Events" to tell process "Sublime Text"
```
What is this action supposed to do? Selecting the previous editor tab?

larsblumberg commented on Application Actions/Sublime Text/1.0/Undo.scpt:
```diff
@@ -0,0 +1,3 @@
+tell application "System Events" to tell process "Sublime Text"
+	key code 6 using {command down}
```
Language dependent keycode

larsblumberg commented on Application Actions/VLC/1.0/default.json:
```diff
@@ -0,0 +1,11 @@
+  "SwipeRight"	:	"Next",
+  "FlyLeft"		:	"Previous",
+  "FlyRight"	:	"Next"
+}
```
Can you please properly format this file without tabs and left-aligning the actions? Thanks!

larsblumberg commented:
@ejwill: Thanks a lot for your effort – I've figured out a few issues – can you please discuss/improve them? Thanks! Lars

ejwill committed:
update to bring some files inline to Lar's issues

fixed some of the issues brought up in the pull request

ejwill commented:
@larsblumberg I believe I fixed most of the issues you brought up.  The only issue I was not able to fix was to figure out how to interact with Sublime Text through the menu.  I tried `tell process "Sublime Text"
    click menu item "Toggle Comment" of menu 1 
        of menu item item "Comment" of menu 1 
        of menu item "Edit" of menu 1
end tell`
but it did not work.  Perhaps we could do separate actions for the different language keyboards?  If not we can pull Sublime Text from the pull request.

larsblumberg commented:
Great @ejwill, thanks a lot for your update. I'll have a look into how to click menu items with AppleScripts and will let you know here.

larsblumberg commented:
@ejwill Sorry for responding so late! Here's how to click menu items, exemplary for Sublime Text and "Toggle Comment":

```
tell application "System Events"
    tell process "Sublime Text"
        click menu item "Toggle Comment" of menu 1 of menu item "Comment" of menu 1 of menu bar item "Edit" of menu bar 1
    end tell
end tell
```

Can you please update your Sublime Text actions that use language dependent keyboard short cuts to use menu item clicks instead?

If you have time, please keep the existing formatting of all `default.json` – You use a different indenting then the current formatting standard. You can [check the diff](https://github.com/getsenic/nuimo-app-osx-actions/pull/12/files) to see the difference.

Then we're ready to merge I guess – Thanks a lot!

ejwill committed:
Merge remote-tracking branch 'getsenic/master'

ejwill committed:
Fixed Sublime Text, Pulled Pixelmater out

I made it so sublime text actions are done through the menu.  Removed
the redo and undo actions because I could not do them through the menu.
 I pulled Pixelmatr out of the pull request. I’ll fix the those actions
later and issue another pull request later.  I also believe the default
files are in the format you want Lars.

ejwill commented:
@larsblumberg take a look now.  I believe I took care of all of your issues.

larsblumberg commented:
Thank you so much for your Pull Request, we can merge it now.
